### ISSUE TITLE
Flowlines `dx_meter` should be a dynamic attribute

### ISSUE THREAD

**Issue by matthiasdusch** [OGGM/oggm#638]
As mentioned by @fmaussion in #624, `Flowline.dx_meter` can and should be dynamic attribute.

**Comment by fmaussion** [comment]
Yes! I'm just reluctant to change it now because it might make existing pickle files un-openable :sob: 

I'd rather stick with it and change this together with a larger refactoring when necessary

**Comment by fmaussion** [comment]
(I said it "might" because I actually don't know)

**Comment by fmaussion** [comment]
A good test would be this here: https://github.com/OGGM/oggm/blob/a74695fcaba0fc50580109bb578ff64df51b3f62/oggm/tests/test_workflow.py#L289-L305 (currently marked as "allowed to fail").

If this test passes after the change I'm happy to merge it
